api/v1alpha1/stage_types.go

```diff
@@ -22,6 +22,9 @@ const (
 	StagePhasePromoting StagePhase = "Promoting"
 	// StagePhaseVerifying denotes a Stage that is currently being verified.
 	StagePhaseVerifying StagePhase = "Verifying"
+	// StagePhaseFailed denotes a Stage that is in a failed state. For example,
+	// the Stage may have failed to promote or verify its Freight.
+	StagePhaseFailed StagePhase = "Failed"
```

Comment:
@krancour happy for this to be named differently, but I think this accomplishes what you were after.